src/vector/focus.rs

```diff
@@ -257,9 +259,18 @@ where
 }
 
 pub struct TreeFocus<A> {
+    /// A clone of the Vector's internal tree that this focus points to. A clone ensures that we don't require a
+    /// reference to the original tree.
     tree: RRB<A>,
+    /// The view represents the range of the tree that this TreeFocus can see. The view can be narrowed by calling
+    /// either the narrow or split_at methods.
     view: Range<usize>,
+    /// The tree version of the Vector is represented as the concatenation of 2 chunks, followed by the tree root,
+    /// followed by 2 chunks. The middle_range refers to the range of the Vector that the tree covers.
     middle_range: Range<usize>,
+    /// This implementation of a focus stores only a single chunk for the Vector. This chunk can refer to one of the 4
+    /// chunks front/back chunks or one of the leaves of the tree. The target_ptr is the pointer to the actual chunk
+    /// in question. The target_range is the range that the chunk represents.
```

Comment:
It gets complicated. A focus might cover multiple nodes in the tree (worse some of them can be partial) so you need to keep a list of covered subtree pointers. Secondly, one of the optimisations that are applied to the structure is to keep 2 chunks on each end of the tree (outside of the tree) so you can speed up access to the front and back. This complicates things further.

One thing that might be worth noting, this implementation is different to ones I have read about before (and how I implemented it). In my implementation, I keep the entire root-to-leaf path in the Focus so you can re-use a bit of the traversal when you need to move to the next leaf. This complicates things because when a Focus is split, the resultant Foci might need to share nodes. The logic of the Focus data structure needs to ensure that you cannot write to bits of a shared node that you don't own.

The Focus here is much simpler, it only keeps the leaf that is in focus in the Focus. When the next leaf is required it does a full root-to-leaf traversal again.